src/node_file.cc

```diff
@@ -780,6 +780,32 @@ void AfterStat(uv_fs_t* req) {
   }
 }
 
+void OpenFileAfterOpen(uv_fs_t* open_req) {
+  FSReqBase* req_wrap = FSReqBase::from_req(open_req);
+  int result = static_cast<int>(open_req->result);
+
+  FS_ASYNC_TRACE_END1(
+      open_req->fs_type, req_wrap, "result", result)
+
+  if (result < 0) {
+    FSReqAfterScope after(req_wrap, open_req);
+    req_wrap->Reject(
+      UVException(req_wrap->env()->isolate(), result, "open", nullptr, open_req->path));
+
+    req_wrap = nullptr;
+    return;
+  }
+
+  char* fd = static_cast<char *>(open_req->data);
+  fd[0] = result;
```

Comment:
`char` is not a viable type for storing a file descriptor since it has a much smaller range